test/utils.spec.js

```diff
@@ -58,4 +59,151 @@ describe('utils', function () {
+      var queryParams = {
+        service_id: service_id,
+        plan_id: plan_id,
+        parameters: parameters,
+        previous_values: {
+          plan_id: plan_id,
+          service_id: service_id
+        }
+      };
+      expect(utils.isNotPlanUpdate(queryParams)).to.be.true;
+    });
+    it('plan update (previous_values.plan_id and plan_id are not same), should return false', function () {
+      var queryParams = {
+        service_id: service_id,
+        plan_id: plan_id_update,
+        parameters: parameters,
+        previous_values: {
+          plan_id: plan_id,
+          service_id: service_id
+        }
+      };
+      expect(utils.isNotPlanUpdate(queryParams)).to.be.false;
+    });
+  });
+
+  describe('#isSameSkuUpdate', function () {
+    /* jshint expr:true */
+    const service_id = '24731fb8-7b84-4f57-914f-c3d55d793dd4';
+    const plan_id = 'bc158c9a-7934-401e-94ab-057082a5073f'; // name: 'v1.0-xsmall'
+    const plan_id_major_version_update = 'gd158c9a-7934-401e-94ab-057082a5073e'; // name: 'v2.0-xsmall'
+    const plan_id_update = 'bc158c9a-7934-401e-94ab-057082a5073e'; // name: 'v1.0-small'
+    const parameters = {
+      foo: 'bar'
+    };
+    it('same sku update (previous_values.plan_id not passed), should return true', function () {
+      var queryParams = {
+        service_id: service_id,
+        plan_id: plan_id_update,
+        parameters: parameters,
+        previous_values: {
+          service_id: service_id
+        }
+      };
+      expect(utils.isSameSkuUpdate(queryParams, catalog)).to.be.true;
+    });
+    it('same sku update (plan_id not passed), should return true', function () {
+      var queryParams = {
+        service_id: service_id,
+        parameters: parameters,
+        previous_values: {
+          plan_id: plan_id,
+          service_id: service_id
+        }
+      };
+      expect(utils.isSameSkuUpdate(queryParams, catalog)).to.be.true;
+    });
+    it('same sku update (previous_values.plan_id sku and plan_id sku are same), should return true', function () {
+      var queryParams = {
+        service_id: service_id,
+        plan_id: plan_id_major_version_update,
+        parameters: parameters,
+        previous_values: {
+          plan_id: plan_id,
+          service_id: service_id
+        }
+      };
+      expect(utils.isSameSkuUpdate(queryParams, catalog)).to.be.true;
+    });
+    it('same sku update (previous_values.plan_id sku and plan_id sku are not same) should return false', function () {
+      var queryParams = {
+        service_id: service_id,
+        plan_id: plan_id_update,
+        parameters: parameters,
+        previous_values: {
+          plan_id: plan_id,
+          service_id: service_id
+        }
+      };
+      expect(utils.isSameSkuUpdate(queryParams, catalog)).to.be.false;
+    });
+  });
+
+  describe('#getSkuNameForPlan', function () {
+    it('should return sku names for plan', function () {
```

Comment:
is `-` part of sku ? ideally sku name will only be xsmall , small etc.. right ?

Reply:
We do sku regex matching. We added '-' in sku so that the sku for plans like v1.0-small and v1.0-xsmall (since small is substring of xsmall) do not match.

Reply:
ok , got it. thanks !